src/main/java/com/google/sps/database/DatabaseImpl.java

```diff
@@ -0,0 +1,279 @@
+
+    //================================================================================
+    // General Test Function (not used for official testing)
+    //================================================================================
+    
+    public void test() {
+        
+        Map<String, List<Long>> hm = new HashMap<String, List<Long>>();
+
+        String caption = "John Cena";
+        List<Long> timestamps = new ArrayList<>();
+        timestamps.add(42L);
+        timestamps.add(365L);
+        timestamps.add(31415L);
+        hm.put(caption, timestamps);
+
+        String caption2 = "Dwayne 'the Rock' Johnson";
+        List<Long> timestamps2 = new ArrayList<>();
+        timestamps2.add(0L);
+        timestamps2.add(123L);
+        timestamps2.add(404L);
+        hm.put(caption2, timestamps2);
+
+        Entity myEnt = new Entity(V_KIND,"ncbb5B85sd2");
+
+        datastore.put(myEnt);
+
+        List<Entity> le = new ArrayList<>();
+
+        for (Map.Entry<String, List<Long>> me : hm.entrySet()) {
+            Entity tempEnt = new Entity(C_KIND, me.getKey(), myEnt.getKey());
+            tempEnt.setProperty(T_KIND, me.getValue());
+            datastore.put(tempEnt);
+            le.add(tempEnt);
+        }
+        
+        for (Entity e : le) {
+            System.out.println("Parent: " + e.getParent());
+        }
+
+    }
+
+    //================================================================================
+    // Public Interface Functions (see DatabaseInterface.java for official descriptions)
+    //================================================================================    
+
+    // add a video to the database
+    public int addVideo(String videoID, String metadata) {
+        Entity vidEnt, metaEnt;
+        try {
+            vidEnt = new Entity(V_KIND, videoID);
+            datastore.put(vidEnt);
+        } catch (Exception e) {
+            System.out.println("Failed to add video to the database");
+            return -1;
+        }
+
+        try {
+            metaEnt = new Entity(M_KIND, metadata, vidEnt.getKey());
+            datastore.put(metaEnt);
+        } catch (Exception e) {
+            System.out.println("Failed to add metadata to the database");
+            return -2;
+        }
+        
+        return 0;
+    }
+
+    // add a keyphrase + timestamp pair to a particular video in the db
+    public int addClause(String videoID, String keyword, List<Long> timestamps) {
+        Entity vidEnt;
+        if ((vidEnt = getVideo(videoID)) == null) {
+            System.out.println("Failed to retrieve " + videoID + " from database");
+            return -1;
+        }
+
+        try {
+            Entity keyEnt = new Entity(C_KIND, keyword, vidEnt.getKey());
+            keyEnt.setProperty(T_KIND, timestamps);
+            datastore.put(keyEnt);
+        } catch (Exception e) {
+            System.out.println("Failed to add " + keyword + " to database");
+            return -2;
```

Comment:
Try to avoid using magic numbers. You could:
1) Define error code or
2) Create an error exception class and throw the error.